File: app/src/main/java/bobrchess/of/by/belaruschess/presenter/RegistrationPresenter.java
package bobrchess.of.by.belaruschess.presenter;

import android.view.View;

import bobrchess.of.by.belaruschess.dto.UserDTO;
import bobrchess.of.by.belaruschess.view.activity.RegistrationContractView;

public interface RegistrationPresenter extends BasePresenter{
    void registration(UserDTO userDTO);
    void loadCoaches();
    void loadRanks();
    void loadCountries();
    void attachView(RegistrationContractView activity);
    void attachViewComponent(View view);
    void setConnectivityStatus(Integer status);
    void setSelectedRankIndex(Integer selectedRankIndex);
    void setSelectedCountryIndex(Integer selectedCountryIndex);
    void setSelectedCoachIndex(Integer selectedCoachIndex);
    void setSelectedGenderIndex(Integer selectedCoachIndex);
    void loadSpinnersData();
}

File: app/src/main/java/bobrchess/of/by/belaruschess/presenter/RegistrationPresenter.kt
package bobrchess.of.by.belaruschess.presenter

import android.view.View
import bobrchess.of.by.belaruschess.dto.extended.ExtendedUserDTO

import bobrchess.of.by.belaruschess.view.activity.RegistrationContractView

interface RegistrationPresenter : BasePresenter {
    fun registration(userDTO: ExtendedUserDTO, userImageUri: String?)
    fun loadCoaches()
    fun loadRanks()
    fun loadCountries()
    fun spinnersAreLoaded(): Boolean
    fun attachView(activity: RegistrationContractView)
    fun attachViewComponent(view: View)
    fun setSelectedRankIndex(selectedRankIndex: Int)
    fun setSelectedCountryIndex(selectedCountryIndex: Int)
    fun setSelectedCoachIndex(selectedCoachIndex: Int)
    fun setSelectedGenderIndex(selectedGenderIndex: Int)
    fun loadSpinnersData()
}